body.go

```diff
@@ -0,0 +1,64 @@
+package main
```

Comment:
this might be a misunderstanding of how package naming and imports work in go.  My understanding is that if someone imports `github.com/taskcluster/taskcluster-lib-artifact-go` that they'd get the `main` package, and that if I had something like `package artifact`, then someone would have to import `github.com/taskcluster/taskcluster-lib-artifact-go/artifact`.  Am I misunderstanding that?

Reply:
Yes, when you import `github.com/taskcluster/taskcluster-lib-artifact-go` the master branch of this repo will be checked out to `${GOPATH}/src/github.com/taskcluster/taskcluster-lib-artifact-go`. Then the `*.go` files in that directory will be read, and it will be necessary for them all to have a `package <x>` declaration in them. The value of `<x>` needs to be the same in all the `*.go` files in order for the compile to succeed. The name here can be different to the enclosing directory name, but by convention it is usually the same - not, the package name being the same as the directory name is _not_ enforced. After you have imported this package, the code that uses code in this package will need to refer to exported identifiers as `<x>.<identifier>`. For example, if you name your package with `package artifact` and provide a function `NewBody` in it, then code that calls it will need to look as follows:

```go
package otherpackage

import (
    "github.com/taskcluster/taskcluster-lib-artifact-go"
)

func SomeFunction() {
    body := artifact.NewBody()
}
```

Note, it is tricky to see from this code that `artifact` is the package name of the package located at `"github.com/taskcluster/taskcluster-lib-artifact-go"` which is why by convention we name the package the same as the directory it is located in. With a directory that has hyphens (`taskcluster-lib-artifact-go`) we cannot use the same name, since hyphens are not allowed in package names. In order to make it easier to see the connection between `taskcluster-lib-artifact-go` and `artifact` you can simply name the package at import. This also solves the problem if you import two different packages that clash on package name, e.g.:

```go
package otherpackage

import (
    artifact "github.com/taskcluster/taskcluster-lib-artifact-go"
    artifact2 "github.com/some/package/also/called/artifact"
)

func SomeFunction() {
    body := artifact.NewBody()
    otherArtifact := artifact2.New("apples") // some function in the other artifact package
}
```